src/test/java/server/NameGeneratorTest.java

```diff
@@ -0,0 +1,38 @@
+package server;
+
+import org.junit.Test;
+import server.ftpcommands.utils.NameGenerator;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class NameGeneratorTest {
+
+    private List<String> existingNames = Arrays.asList("hello.txt", "hello-1.txt", "hello-2.txt", "hello.world.txt");
+    private NameGenerator nameGenerator = new NameGenerator(this::isUniqueName);
+
+    @Test
+    public void generate() {
+        String name = nameGenerator.generateUnique("foo.txt");
+        assertEquals("foo.txt", name);
+    }
+
+    @Test
+    public void generateUnique() {
+        String name = nameGenerator.generateUnique("hello.txt");
```

Comment:
Maybe easier to read you test 
1/ create a name `generator.addName("bla")`
2/ generated name `generator.generateUnique("bla")`
3/ expect bla to be different because it already existed (1)

This would avoid to jump around the file to understand what's going one